platforms/core-runtime/base-services/src/main/java/org/gradle/internal/reflect/JavaMethod.java

```diff
@@ -116,11 +116,11 @@ public boolean isStatic() {
         return Modifier.isStatic(method.getModifiers());
     }
 
-    public R invokeStatic(Object... args) {
+    public R invokeStatic(@Nullable Object... args) {
         return invoke(null, args);
     }
 
-    public R invoke(@Nullable T target, Object... args) {
+    public R invoke(@Nullable T target, @Nullable Object... args) {
```

Comment:
❓ Same as above, do we have callsites that require the ability to pass `null` here?

Reply:
After taking a quick look, it's hard to say. Most of the callsites indeed seem to provide non-nullable arguments, but not all the code the callsites belong to is non-null annotated.

Moreover, I am not certain if this method (as well as `invokeStatic` above) are not used from some runtime-generated code, given how `invokeStatic` does not have any usages in production code at all.

This is a very generic method that mirrors calling a Java method, which inherently supports getting nulls as arguments. I would err on the side of caution and mark the arguments as nullable. Especially, because they directly go into the actual reflective call.

Reply:
Ok.